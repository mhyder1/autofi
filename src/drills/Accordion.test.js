import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme';

describe.only(`Accordion Component`, () => {
    let sections = [
        {id:1,title:"Potatoes", content:"Cultivated by aliens."},
        {id:2,title:"Onions", content:"Made of shrek"},
        {id:3,title:"Guitars", content:"My partner knows something about this."}];
    
    it('Smoke test',() =>{
        let div = document.createElement('div');
        ReactDOM.render(<Accordion/>,div);
        ReactDOM.unmountComponentAtNode(div);
    });
    
    it("Returns empty list item, if nothing passed.",() => {
        const wrapper = shallow(<Accordion />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });

    it("Renders with no default content expanded.",() => {
        const wrapper = shallow(<Accordion sections={sections}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    });

})
