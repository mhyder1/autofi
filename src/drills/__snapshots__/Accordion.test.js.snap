// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component Renders with no default content expanded. 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Potatoes
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Onions
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Guitars
    </button>
  </li>
</ul>
`;

exports[`Accordion Component Returns empty list item, if nothing passed. 1`] = `<ul />`;
